### ISSUE TITLE
Redis size assessment

### ISSUE THREAD

**Issue by orgalaf** [OptimalBits/bull#1361]
Hi,
Not really a bug rather a request for better docs. Recently tried deploying bull in production over aws elasticache to replace rabbitmq. We estimated the data size to be about 20GB and provisioned a 30GB redis instance. After about 15 mins in production we had about 10GB of data in redis and started getting OOM errors. A bit of research revealed that write heavy applications over redis (as bulljs is) can incur significant memory overhead up to twice the size of the actual data.

Would be good to get some guidelines in the docs about what size of instance to provision for the expected load.

Thanks

**Comment by manast** [comment]
So the overhead of bull is inversely proportional to the size of the data object that you send with every job. If the data object is big, the bull's overhead is minimal. What is your use case, many small jobs or big ones?

**Comment by orgalaf** [comment]
The objects are about 100kb on average. And we probably process about 70k per minute.

**Comment by manast** [comment]
overhead is the extradata that decorates a job, but that is just a couple of hundred of bytes, so negligible in this case. However, 70K * 100K is 7Gb per minute, I wonder if these numbers are correct?

**Comment by orgalaf** [comment]
Yep sorry my bad was looking at the wrong charts. It's 3.5K messages/minute. But the message size I'm pretty sure about. Possibly it could be a bit less. So about 230MB/min of incoming messages, most of which should get cleared out as they finish processing. All except for those that fail and get retried or fail completely and stay in the failed queue.
